robot-core/src/main/java/org/obolibrary/robot/template/Template.java

```diff
@@ -0,0 +1,1852 @@
+    axioms.add(ax);
+
+    String classType = null;
+    if (classTypeColumn != -1) {
+      try {
+        classType = row.get(classTypeColumn);
+      } catch (IndexOutOfBoundsException e) {
+        // do nothing
+      }
+    }
+    if (classType == null || classType.trim().isEmpty()) {
+      classType = "subclass";
+    } else {
+      classType = classType.trim().toLowerCase();
+    }
+
+    Set<OWLClassExpression> expressions = new HashSet<>();
+    for (int column = 0; column < templates.size(); column++) {
+      String template = templates.get(column);
+      String value = null;
+      try {
+        value = row.get(column);
+      } catch (IndexOutOfBoundsException e) {
+        // do nothing
+      }
+
+      if (value == null || value.trim().isEmpty()) {
+        continue;
+      }
+
+      if (template.startsWith("A")) {
+        // Handle class annotations
+        Set<OWLAnnotation> annotations = getAnnotations(template, value, row, column);
+        for (OWLAnnotation annotation : annotations) {
+          axioms.add(dataFactory.getOWLAnnotationAssertionAxiom(iri, annotation));
+        }
+      } else if (template.startsWith("C") && !template.startsWith("CLASS_TYPE")) {
+        // Handle class logic
+        expressions.addAll(TemplateHelper.getClassExpressions(parser, template, value));
+        switch (classType) {
+          case "subclass":
+            addSubClassAxioms(cls, expressions, row, column);
+            break;
+          case "equivalent":
+            addEquivalentClassesAxioms(cls, expressions, row, column);
+            break;
+          case "disjoint":
+            addDisjointClassAxioms(cls, expressions, row, column);
+            break;
+          default:
+            // TODO - unknown class type
+            throw new Exception();
+        }
+      }
+    }
+  }
+
+  /**
+   * Given an OWLClass, a set of OWLClassExpressions, and the row containing the class details,
+   * generate subClassOf axioms for the class where the parents are the class expressions. Maybe
+   * annotate the axioms.
+   *
+   * @param cls OWLClass to create subClassOf axioms for
+   * @param expressions set of parent OWLClassExpressions
+   * @param row list of template values for given class
+   * @param column column number of logical template string
+   * @throws Exception on issue getting axiom annotations
+   */
+  private void addSubClassAxioms(
+      OWLClass cls, Set<OWLClassExpression> expressions, List<String> row, int column)
+      throws Exception {
+    // Maybe get an annotation on the expression
+    String nextTemplate;
+    try {
+      nextTemplate = templates.get(column + 1);
+    } catch (IndexOutOfBoundsException e) {
+      nextTemplate = null;
+    }
+
+    // Support for annotations on: superclasses, domains, and ranges
+    Set<OWLAnnotation> axiomAnnotations = new HashSet<>();
+    if (nextTemplate != null && !nextTemplate.trim().isEmpty() && (nextTemplate.startsWith(">C"))) {
```

Comment:
">C" seems wrong to me. Shouldn't it be ">A"? I see the same thing below. We definitely need a test that covers these annotations.

Reply:
I left a comment in the PR in regards to this.

Reply:
Changed to all `>A*`. `>C*` can still be used but it will log a warning to use `>A*` instead.